### PR TITLE
Groups

### PR DESCRIPTION
**Pull request by anitaschiller**
Branch for user stories: Add groups, delete groups & filter groups

I added a "NewGroup" component to the Add page that makes creating a new group possible. Already existing groups can be deleted on the Home page if there aren't any remaining members in the group. At the top of the page there's a group filter now so that the user can easily find the members of a specific group.

Next step: Add searchbar

Thank you!! :-)

### CODE REVIEW

**Review by m-wiemers — APPROVED** [review]
gj 🚀

File: client/src/App.js [hunk 1]
```diff
@@ -56,16 +64,45 @@ function App() {
...
 
+  function addGroup(groupToAdd) {
+    setAvailableGroups([...availableGroups, groupToAdd]);
+  }
+
+  function deleteGroup(groupToDelete) {
+    const groupMembers = members.filter(
+      (member) => member.group === groupToDelete
+    );
+    console.log('groupMembers', groupMembers);
```
Comment (miriamhaenle) [inline client/src/App.js:75]:
```suggestion
```

File: client/src/components/ErrorMessage.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import styled from 'styled-components';
+
+export default function ErrorMessage({ text }) {
+  return (
+    <>
```
Comment (miriamhaenle) [inline client/src/components/ErrorMessage.js:5]: I think you don't need the React Fragments here as the <Error> is the only return statement from the function

File: client/src/pages/Home.js [hunk 3]
```diff
@@ -14,20 +26,86 @@ export default function Home({ members, onOpenModal }) {
...
+        </select>
+        <span onClick={() => setRenderedGroups(availableGroups)}>&times;</span>
+      </GroupFilter>
+      {renderedGroups.map((group) => (
+        <GroupWrapper>
+          <GroupHeadline>
+            {group}
+            <Delete onClick={() => deleteGroup(group)}>&times;</Delete>
+          </GroupHeadline>
+          {cantDeleteGroup === group && (
```
Comment (miriamhaenle) [inline client/src/pages/Home.js:57]: ahh! now I get the naming 💡

File: client/src/pages/Home.js [hunk 4]
```diff
@@ -1,8 +1,20 @@
...
+export default function Home({
+  members,
+  onOpenModal,
+  availableGroups,
+  deleteGroup,
+  cantDeleteGroup,
+}) {
   const orderedMembers = members.slice().sort(compareFirstName);
+  const [groupValue, setGroupValue] = useState('');
+  const [renderedGroups, setRenderedGroups] = useState(availableGroups ?? []);
```
Comment (miriamhaenle) [inline client/src/pages/Home.js:17]: Setting the initial value to 'availableGroups' like you do will cause errors when you want to update the component immediately (e.g. after deleting a group on the home screen). In order to do so you need to actually update `renderedGroups` with a useEffect hook: 

`const [renderedGroups, setRenderedGroups] = useState([])

useEffect(() => {
setRenderedGroups(availableGroups)
}, [availableGroups])
`

File: client/src/App.js [hunk 5]
```diff
@@ -15,11 +15,19 @@ function App() {
   const [members, setMembers] = useState(loadFromLocal('members') ?? []);
   const [isShown, setIsShown] = useState(false);
   const [remainingMembers, setRemainingMembers] = useState(members);
+  const [availableGroups, setAvailableGroups] = useState(
+    loadFromLocal('groups') ?? []
+  );
+  const [cantDeleteGroup, setCantDeleteGroup] = useState('');
```
Comment (miriamhaenle) [inline client/src/App.js:21]: I was put off a bit by the naming. Maybe you can think again to find a name that is more precise. 
What do you need this additional state for?

File: client/src/App.js [hunk 6]
```diff
@@ -56,16 +64,45 @@ function App() {
     }
   }
 
+  function addGroup(groupToAdd) {
+    setAvailableGroups([...availableGroups, groupToAdd]);
+  }
+
+  function deleteGroup(groupToDelete) {
```
Comment (miriamhaenle) [inline client/src/App.js:71]: The delete function does sucessfully delete the group however the the group only disappears on reload on the home screen at the moment. This is related to how you render the groups in the home component (I added a comment there as well)

**Review by miriamhaenle — CHANGES_REQUESTED** [review]
Hi Anita, 
nicely done! 
I added a few suggestions on how you render the Home component. 

Happy easter 🐰

File: client/src/pages/Home.js (hunk 4 above)
Comment (anitaschiller) [reply client/src/pages/Home.js:17]: I realized that too (quite late unfortunately) when I implemented the Searchbar and fixed it in the next branch. Your solution however looks much more elegant, so I will see if I can include it in the searchbar branch :-).

File: client/src/pages/Home.js (hunk 3 above)
Comment (anitaschiller) [reply client/src/pages/Home.js:57]: 😃 it's probably not optimal - I will think about the naming again

File: client/src/App.js (hunk 1 above)
Comment (anitaschiller) [reply client/src/App.js:75]: 🙈

**Pull request merged**